src/libclipper/test/serialization_test.cpp

```diff
@@ -272,4 +272,130 @@ namespace {
+    // Removing the last element of the second vector renders the first two vectors
+    // distinct, so they should have different hashes
+    ASSERT_NE(FloatVector(float_hash_vecs[0]).hash(), FloatVector(float_hash_vecs[1]).hash());
+    float_hash_vecs[1].push_back(500);
+    // Adding the element 500.0, which is not present in the first vector, to the second vector
+    // leaves the first two vectors distinct, so they should have different hashes
+    ASSERT_NE(FloatVector(float_hash_vecs[0]).hash(), FloatVector(float_hash_vecs[1]).hash());
+    std::reverse(float_hash_vecs[2].begin(), float_hash_vecs[2].end());
+    // Reversing the third vector, which was initially the reverse of the first vector,
+    // renders the first and third vectors identical, so they should have the same hash
+    ASSERT_EQ(FloatVector(float_hash_vecs[0]).hash(), FloatVector(float_hash_vecs[2]).hash());
+  }
+
+  TEST(SerializationTests, DoubleVectorsHashCorrectly) {
+    // Obtains 3 vectors containing double intepretations of the integers 0-99.
+    // The first two are identical, and the third is reversed
+    std::vector<std::vector<double>> double_hash_vecs = get_primitive_hash_vectors<double>();
+    ASSERT_EQ(DoubleVector(double_hash_vecs[0]).hash(), DoubleVector(double_hash_vecs[1]).hash());
+    ASSERT_NE(DoubleVector(double_hash_vecs[0]).hash(), DoubleVector(double_hash_vecs[2]).hash());
+    double_hash_vecs[1].pop_back();
+    // Removing the last element of the second vector renders the first two vectors
+    // distinct, so they should have different hashes
+    ASSERT_NE(DoubleVector(double_hash_vecs[0]).hash(), DoubleVector(double_hash_vecs[1]).hash());
+    double_hash_vecs[1].push_back(500);
+    // Adding the element 500.0, which is not present in the first vector, to the second vector
+    // leaves the first two vectors distinct, so they should have different hashes
+    ASSERT_NE(DoubleVector(double_hash_vecs[0]).hash(), DoubleVector(double_hash_vecs[1]).hash());
+    std::reverse(double_hash_vecs[2].begin(), double_hash_vecs[2].end());
+    // Reversing the third vector, which was initially the reverse of the first vector,
+    // renders the first and third vectors identical, so they should have the same hash
+    ASSERT_EQ(DoubleVector(double_hash_vecs[0]).hash(), DoubleVector(double_hash_vecs[2]).hash());
+  }
+
+  TEST(SerializationTests, ByteVectorsHashCorrectly) {
+    // Obtains 3 vectors containing byte intepretations of the integers 0-99.
+    // The first two are identical, and the third is reversed
+    std::vector<uint8_t> byte_vec_1;
+    for(int i = 0; i < 100; i++) {
+      uint8_t* bytes = reinterpret_cast<uint8_t*>(i);
+      for(int j = 0; j < static_cast<int>(sizeof(int) / sizeof(uint8_t)); j++) {
+        byte_vec_1.push_back(bytes[j]);
+      }
+    }
+    std::vector<uint8_t> byte_vec_2 = byte_vec_1;
+    std::vector<uint8_t> byte_vec_3 = byte_vec_1;
+    std::reverse(byte_vec_3.begin(), byte_vec_3.end());
```

Comment:
I was treating byte vectors differently because statically casting an integer to a byte causes a loss of information. For the purposes of a test, it doesn't seem that this should matter. Taking your suggestion, I've opted to call `get_primitive_hash_vectors` with `uint8_t` as a parameter.